### PR TITLE
Run slack notification via active jobs using sidekiq

### LINKED ISSUE #660: Runs Slack Notification in a job on sidekiq
For while, the app is runing on heroku, but I think it's necessary to move on the slack notifications to run on a background service using sidekiq.
The integration with slack I has been created on that issue: 
https://github.com/espoo-dev/espoo-dev/issues/656

### PR DESCRIPTION
**Pull request by maniSHarma7575**
fix #660 

# Run slack notification via Active jobs using Sidekiq.

## Type of changes

Please delete options that are not relevant.

- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Breaking change**

---

## Checklist:

- [x] My code is understandable
- [x] My code follows the styleguide of this project
- [x] My changes don't generate new warnings/errors
- [x] I have performed a self-review of my own code (build/lint/tests)

  ### Backend

  Please delete options that are not relevant.

  - [x] **Model testing.**
  - [ ] **Request testing and swagger doc update.**
  - [ ] **System testing.**
  - [ ] **No tests required.**

  ### Frontend

  - Components:

    - [ ] I have implemented/updated unit tests
    - [ ] I have implemented/updated Storybook documentation

  - Pages:
    - [ ] I have implemented/updated E2E tests

### Resources:

1. Redis is required to be installed after merging the changes.

```
heroku addons:create heroku-redis:hobby-dev
heroku ps:scale worker=1
```

Link: [https://www.bootrails.com/blog/rails-sidekiq-tutorial/](https://www.bootrails.com/blog/rails-sidekiq-tutorial/)

### CODE REVIEW

**Comment by maniSHarma7575** [comment]
@valmirosjunior Ready for review.

File: backend/Procfile [hunk 1]
```diff
@@ -1,2 +1,3 @@
 release: bundle exec rake db:migrate db:seed
 web: bundle exec rails server -p $PORT
+worker: bundle exec sidekiq -C config/sidekiq.yml
```
Comment (valmirosjunior) [inline backend/Procfile:3]:
```suggestion
worker: bundle exec sidekiq -C config/sidekiq.yml

```

File: backend/Procfile.dev [hunk 2]
```diff
@@ -0,0 +1,2 @@
+web: bundle exec rails s -b 0 --port=3000
+worker: bundle exec sidekiq -C config/sidekiq.yml
```
Comment (valmirosjunior) [inline backend/Procfile.dev:2]:
```suggestion
worker: bundle exec sidekiq -C config/sidekiq.yml

```

File: backend/config/initializers/sidekiq.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+db_name    = ENV.fetch('REDIS_DB_NAME', '1')
+redis_url  = "redis://#{redis_host}:#{redis_port}/#{db_name}"
+
+Sidekiq.configure_server do |config|
+  config.redis = { url: redis_url }
+end
+
+Sidekiq.configure_client do |config|
+  config.redis = { url: redis_url }
+end
```
Comment (valmirosjunior) [inline backend/config/initializers/sidekiq.rb:12]:
```suggestion
end

```

File: backend/config/sidekiq.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+development:
+  :concurrency: 5
+
+production:
+  :concurrency: 10
+
+:max_retries: 1
+
+:queues:
+  - default
```
Comment (valmirosjunior) [inline backend/config/sidekiq.yml:10]:
```suggestion
  - default
  
```

File: backend/spec/jobs/notification_slack_job_spec.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe NotificationSlackJob, type: :job do
```
Comment (valmirosjunior) [inline backend/spec/jobs/notification_slack_job_spec.rb:3]: In general, I liked the tests, but I think that it's also important to check if, the `SlackService`, has been called, with correct params, what do you think about it?

**Review by valmirosjunior — CHANGES_REQUESTED** [review]

File: backend/spec/jobs/notification_slack_job_spec.rb (hunk 5 above)
Comment (maniSHarma7575) [reply backend/spec/jobs/notification_slack_job_spec.rb:3]: Yes, added a test to check if slack service call method has been called.

File: backend/config/sidekiq.yml (hunk 4 above)
Comment (maniSHarma7575) [reply backend/config/sidekiq.yml]: Updated

File: backend/config/initializers/sidekiq.rb (hunk 3 above)
Comment (maniSHarma7575) [reply backend/config/initializers/sidekiq.rb]: Updated

File: backend/Procfile.dev (hunk 2 above)
Comment (maniSHarma7575) [reply backend/Procfile.dev]: Updated

File: backend/Procfile (hunk 1 above)
Comment (maniSHarma7575) [reply backend/Procfile]: Updated

**Review by edimossilva — APPROVED** [review]
@maniSHarma7575 and @valmirosjunior , great work, nice to see you collaborating.
LGTM!

**Review by valmirosjunior — APPROVED** [review]
Great job @maniSHarma7575 .

**Pull request merged**